tests/unit/test_ballot_box.py

```diff
@@ -152,3 +157,78 @@ def test_spoil_ballot(self):
         self.assertIsNone(
             accept_ballot(data, BallotBoxState.CAST, internal_manifest, context, store)
         )  # cannot cast a ballot already spoiled
+
+    def test_ballot_box_submit_ballot(self):
+        # Arrange
+        keypair = elgamal_keypair_from_secret(int_to_q(2))
+        manifest = election_factory.get_fake_manifest()
+        internal_manifest, context = election_factory.get_fake_ciphertext_election(
+            manifest, keypair.public_key
+        )
+        store = DataStore()
+        source = election_factory.get_fake_ballot(internal_manifest)
+        self.assertTrue(source.is_valid(internal_manifest.ballot_styles[0].object_id))
+
+        # Act
+        data = submit_ballot(internal_manifest, BallotBoxState.SPOILED)
```

Comment:
The `internal_manifest` seems to be confused with `source` in these tests.

The `internal_manifest` is just an internal representation of the `manifest` which is HOW the ballots should look but not a ballot. 

Check the types. The `get_fake_ballot` should give you a ballot.